modules/adagioBidAdapter.js

```diff
@@ -295,29 +306,38 @@ export const spec = {
   supportedMediaType: SUPPORTED_MEDIA_TYPES,
 
   isBidRequestValid: function(bid) {
-    top.ADAGIO = top.ADAGIO || {};
-    top.ADAGIO.adUnits = top.ADAGIO.adUnits || {};
-    top.ADAGIO.pbjsAdUnits = top.ADAGIO.pbjsAdUnits || [];
     const { adUnitCode, auctionId, sizes, bidder, params } = bid;
     const { organizationId, site, placement, pagetype, adUnitElementId } = bid.params;
-    const isValid = !!(organizationId && site && placement && pagetype && adUnitElementId && document.getElementById(adUnitElementId) !== null);
-    const tempAdUnits = top.ADAGIO.pbjsAdUnits.filter((adUnit) => adUnit.code !== adUnitCode);
-    tempAdUnits.push({
-      code: adUnitCode,
-      sizes,
-      bids: [{
-        bidder,
-        params
-      }]
-    });
-    top.ADAGIO.pbjsAdUnits = tempAdUnits;
+    let isValid = false;
 
-    if (isValid === true) {
-      top.ADAGIO.adUnits[adUnitCode] = {
-        auctionId: auctionId,
-        pageviewId: _getPageviewId()
-      };
+    try {
+      if (window.top.location.href) {
```

Comment:
Je préfèrerais une méthode commune pour ce `if`, qui est fait deux fois dans l'adapter.
Et que le code fasse une sortie de la fonction ASAP, plutot que tout mettre dans un `if` et indenter.

Reply:
Plus généralement, je me dis que le check-and-abord de la safe iframe devrait être mis dans toutes les functions appelées par Prebid.js, par sécurité.
Mais peut-être qu'on veut garder des méthodes accessibles, comme le user-sync, si ça aurait un sens.

Reply:
En effet c'est plus élégant d'avoir un canAccessTopWindow()

Pour l'autre question en effet on peut rajouter ce check a chaque méthode de qui tape dans top.window, mais en vrai on ne passera jamais dedans si la bidRequest est invalide.

Reply:
Ok oublie la question de mettre le check à chaque function.